pkg/cmd/component/component/add.go

```diff
@@ -0,0 +1,203 @@
+package component
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/odo/pkg/log"
+	addui "github.com/redhat-developer/kam/pkg/cmd/component/component/ui"
+	"github.com/redhat-developer/kam/pkg/cmd/genericclioptions"
+	"github.com/redhat-developer/kam/pkg/cmd/ui"
+	pipelines "github.com/redhat-developer/kam/pkg/pipelines/component/component"
+	"github.com/redhat-developer/kam/pkg/pipelines/ioutils"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	// AddCompRecommendedCommandName the recommended command name
+	AddCompRecommendedCommandName = "add"
+)
+
+// AddCompParameters encapsulates the parameters for the kam pipelines init command.
+type AddCompParameters struct {
+	*pipelines.CompomemtParameters
+	Interactive bool
+}
+
+const (
+	componentNameFlag    = "component-name"
+	outputFolderNameFlag = "output"
+	applicationNameFlag  = "application-name"
+)
+
+var (
+	addCompExample = ktemplates.Examples(`
+	# Add a new Component to Application
+	# Example: kam component add --output <path to Application folder> --application-name <Application name to add component> --component-name new-component 
+	
+	%[1]s 
+	`)
+
+	addCompLongDesc  = ktemplates.LongDesc(`Add a new Component to the Application`)
+	addCompShortDesc = `Add a new Component`
+)
+
+// Complete completes AddCompParameters after they've been created.
+func (io *AddCompParameters) Complete(name string, cmd *cobra.Command, args []string) error {
+	if cmd.Flags().NFlag() == 0 || io.Interactive {
+		return initiateInteractiveModeComponent(io, cmd)
+	}
+	return initiateNonInteractiveModeComponent(io)
+}
+
+// Checking the mandatory flags & reusing missingFlagErr in .go
+func checkMandatoryFlags(flags map[string]string) error {
+	missingFlags := []string{}
+	mandatoryFlags := []string{componentNameFlag, outputFolderNameFlag, applicationNameFlag}
+	for _, flag := range mandatoryFlags {
+		if flags[flag] == "" {
+			missingFlags = append(missingFlags, fmt.Sprintf("%q", flag))
+		}
+	}
+	if len(missingFlags) > 0 {
+		return missingFlagErr(missingFlags)
+	}
+	return nil
+}
+
+func missingFlagErr(flags []string) error {
+	return fmt.Errorf("required flag(s) %s not set", strings.Join(flags, ", "))
+}
+
+func initiateNonInteractiveModeComponent(io *AddCompParameters) error {
+	mandatoryFlags := map[string]string{componentNameFlag: io.ComponentName, applicationNameFlag: io.ApplicationName, outputFolderNameFlag: io.OutputFolder}
+	if err := checkMandatoryFlags(mandatoryFlags); err != nil {
+		return err
+	}
+	err := ui.ValidateName(io.ApplicationName)
+	if err != nil {
+		return err
+	}
+	err = ui.ValidateName(io.ComponentName)
+	if err != nil {
+		return err
+	}
+	exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), io.OutputFolder)
+	if !exists {
+		return fmt.Errorf("the given path : %s  doesnot exists ", io.OutputFolder)
```

Comment:
"does not exist"

I'm wondering if we could give it more context.  eg.

How about 

"The provided path does not contain the application with the folder name: %s"
"The provided path does not contain the application: %s"

Reply:
Okay. I will change the error statments.